src/cli.rs

```diff
@@ -338,6 +338,17 @@ async fn process_line(readline: Result<String, ReadlineError>, ctx: &mut Context
         Ok(line) if line.trim() == "" => LineResult::Success(line.clone()),
 
         Ok(line) => {
+            // Replace ~ with the address of the home directory.
+            let mut line_string = String::from(line);
+            if line.contains("~") {
+                match dirs::home_dir() {
+                    // unwrap() is fine here since we know home is a valid unicode string.
+                    Some(home) => line_string = line_string.replace("~", home.to_str().unwrap()),
+                    None => panic!("Cannot find home directory"),
```

Comment:
Instead of searching and replacing across the whole string, we should separate how we handle tilde for the two types of commands:

* internal commands - these will say when they expect a path, and we can then replace the ~ in the path.
* external commands - we should process the arguments given to external like you're suggesting, by replacing the tilde

Reply:
If you want, you could do this in stages like you suggest.